pkg/ottl/ottlfuncs/func_is_root_span.go

```diff
@@ -0,0 +1,25 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+)
+
+func NewIsRootSpanFactory() ottl.Factory[ottlspan.TransformContext] {
```

Comment:
Since this function requires a specific context, but can be used in multiple components, not just the transformprocessor, I think it should consider if it should live in `ottlspan`. Downside is that documenting it would be weird.

Reply:
@evan-bradley if you agree I will move it to `ottlspan` package

Reply:
@TylerHelmuth What do you see as the benefit of moving it to `ottlspan`? Even if it's just for spans, it's still primarily a function, so I think the `ottlfuncs` package probably makes more sense. I don't think moving it to `ottlspan` would really change how most OTTL consumers would use it.

I'm not opposed to moving it to `ottlspan`, but I think I would be more surprised to find it in that package than `ottlfuncs` if I was looking through the directories.

Reply:
I prefer it in `ottlfuncs` as well, but wanted to call it out for the record.  

We should consider if our function docs needs some more structure that clearly calls out which context the function is used for.  This is the first time we've had to specify one.

Reply:
@odubajDT could we add a note in the docs for this function stating that it can only be used in the `span` context?

Reply:
yep, it's already added https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33729/files#diff-b6db71fdf946de5520e511604eb21680bf1c80ba5319b92720237745edc00fa1R726

Reply:
My bad, don't know how I missed that. Looks good, thank you.